samples/wifi/monitor/src/main.c

```diff
@@ -328,6 +328,56 @@ static int wifi_set_channel(void)
 	return 0;
 }
 
+static int wifi_set_packet_filter(void)
+{
+	int ret;
+	struct net_if *iface;
+	struct wifi_filter_info filter_info = { 0 };
+	uint32_t filter = 0;
+
+	iface = net_if_get_first_wifi();
+	if (iface == NULL) {
+		LOG_ERR("No Wi-Fi interface found");
+		return -1;
+	}
+
+	filter_info.if_index = net_if_get_by_iface(iface);
+
+	if (IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_ALL)) {
+		filter |= WIFI_PACKET_FILTER_ALL;
+	} else {
+		if (IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CTRL)) {
+			filter |= WIFI_PACKET_FILTER_MGMT;
+		} else if (IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CTRL)) {
+			filter |= WIFI_PACKET_FILTER_CTRL;
+		} else if (IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_DATA)) {
+			filter |= WIFI_PACKET_FILTER_DATA;
+		}
+	}
+
+	_Static_assert(!(IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CUSTOM) &&
```

Comment:
```suggestion
	BUILD_ASSERT(!(IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CUSTOM) &&
```
In general we shouldn't be using APIs/functions that start with _/__, they are mainly for internal use. Though BUILD_ASSERT is a wrapper around _Static_assert, it has a better name and also gives better control (like disable all build asserts)